### PR TITLE
CircleCI 2.0 configuration

### PR DESCRIPTION
**Pull request by szemek**
Fixes https://github.com/github-changelog-generator/github-changelog-generator/issues/681

Passing build(s) on branch used for this PR:
https://circleci.com/gh/szemek/github-changelog-generator/tree/circleci2.0

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    docker:
+       - image: circleci/ruby:2.4.1-node-browsers
+
+    steps:
+      - checkout
+
+      # download and cache dependencies
+      - restore_cache:
+          keys:
+          - dependencies-{{ checksum "Gemfile" }}-{{ checksum "github_changelog_generator.gemspec" }}
```
Comment (olleolleolle) [inline .circleci/config.yml]:
```suggestion
          - dependencies-{{ checksum "Gemfile.lock" }}-{{ checksum "github_changelog_generator.gemspec" }}
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+      - run:
+          name: Install dependencies
+          command: |
+            bundle install --jobs=4 --retry=3 --path vendor/bundle
+
+      - save_cache:
+          paths:
+            - ./vendor/bundle
+          key: dependencies-{{ checksum "Gemfile" }}-{{ checksum "github_changelog_generator.gemspec" }}
```
Comment (olleolleolle) [inline .circleci/config.yml]:
```suggestion
          key: dependencies-{{ checksum "Gemfile.lock" }}-{{ checksum "github_changelog_generator.gemspec" }}
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Ruby CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/language-ruby/ for more details
+#
+version: 2
+jobs:
+  build:
+    docker:
+       - image: circleci/ruby:2.4.1-node-browsers
```
Comment (olleolleolle) [inline .circleci/config.yml]:
```suggestion
       - image: circleci/ruby:2.5.1-node-browsers
```

**Comment by szemek** [comment]
@olleolleolle I started with `Gemfile.lock` but I got error because this file doesn't exists in repository and it is create after `bundle install`
```
Error computing cache key: template: cacheKey:1:16: executing "cacheKey" at <checksum "Gemfile.lo...>: error calling checksum: open /home/circleci/project/Gemfile.lock: no such file or directory
```

I removed caching in the last commit. Installing dependencies is so fast that it will be fine to check all the builds with latest matching gems.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      - run:
+          name: Install dependencies
+          command: |
+            bundle install --jobs=4 --retry=3 --path vendor/bundle
+
+      # run tests
+      - run:
+          name: Run tests
+          command: |
+            bundle exec rspec --color --require spec_helper spec --format progress
```
Comment (olleolleolle) [inline .circleci/config.yml]: Could we have a `bundle exec rubocop` run step?

Comment (olleolleolle) [reply .circleci/config.yml]: (Before the `rspec` step.)

Comment (szemek) [reply .circleci/config.yml]: added in a next commit

File: .circleci/config.yml [hunk 5]
```diff
@@ -17,6 +17,12 @@ jobs:
           command: |
             bundle install --jobs=4 --retry=3 --path vendor/bundle
 
+      # run rubocop checks
```
Comment (olleolleolle) [inline .circleci/config.yml]: Idea: drop all the comments - you already have VERY good `name` properties.

Comment (szemek) [reply .circleci/config.yml]: you're right, those comments are redundant

**Comment by olleolleolle** [comment]
We're getting ready!

**Comment by olleolleolle** [comment]
Is there a neat way to make a build matrix for more than one version?

**Comment by olleolleolle** [comment]
Something like:

http://yaml-online-parser.appspot.com/?yaml=version%3A+2.0%0A%0Ashared%3A+%26shared%0A++++working_directory%3A+~%2Fcircleci-demo-workflows%0A++++steps%3A%0A++++++-+checkout%0A++++++-+run%3A+bundle+install+--path+vendor%2Fbundle%0A++++++-+run%3A+bundle+exec+rake+db%3Acreate+db%3Aschema%3Aload%0A++++++-+run%3A%0A++++++++++name%3A+Run+tests%0A++++++++++command%3A+rake%0A%0Ajobs%3A%0A++%22ruby-2.2%22%3A++%0A++++%3C%3C%3A+*shared%0A++++docker%3A%0A++++++-+image%3A+circleci%2Fruby%3A2.2-node%0A++++++-

**Comment by olleolleolle** [comment]
Scratch that: that's a scope-creep!

**Comment by olleolleolle** [comment]
@szemek Let's _do this_!

**Comment by olleolleolle** [comment]
@szemek Thank you so much for coming in and _fixing this_!

**Pull request merged**